helix-view/src/theme.rs

```diff
@@ -157,25 +161,38 @@ impl Theme {
         self.highlights[index]
     }
 
-    pub fn get(&self, scope: &str) -> Style {
+    pub fn get<T: AsRef<ThemeKey>>(&self, scope: T) -> Style {
         self.try_get(scope).unwrap_or_default()
     }
 
     /// Get the style of a scope, falling back to dot separated broader
     /// scopes. For example if `ui.text.focus` is not defined in the theme,
     /// `ui.text` is tried and then `ui` is tried.
-    pub fn try_get(&self, scope: &str) -> Option<Style> {
-        std::iter::successors(Some(scope), |s| Some(s.rsplit_once('.')?.0))
-            .find_map(|s| self.styles.get(s).copied())
+    pub fn try_get<T: AsRef<ThemeKey>>(&self, scope: T) -> Option<Style> {
+        // Remove generics to make monomorphization less costly
```

Comment:
That's a good point, using `AsRef<ThemeKey>` in so many places might be pretty costly. https://matklad.github.io/2021/09/04/fast-rust-builds.html#compilation-model-monomorphization seems to indicate it's _mainly_ a problem between multiple crates. This might need some profiling?